File: 프로그래머스/lv2/42885. 구명보트/구명보트.js
function solution(people, limit) {
  people.sort((a, b) => a - b); // 오름차순 정렬
  let min = 0;
  let max = people.length - 1;
  let boat = 0;
  while (min <= max) {
    if (people[min] + people[max] <= limit) {
      min++;
    }
    max--;
    boat++;
  }
  return boat;
}

File: 프로그래머스/lv2/42885. 구명보트/구명보트.java
import java.util.*;

class Solution {
    public int solution(int[] people, int limit) {
        Arrays.sort(people);
        int min = 0;
        int max = people.length - 1;
        int boat = 0;
        while(min <= max){
            if(people[min] + people[max] <= limit){
                min ++;
            }
            max--;
            boat++;
        }
        return boat;
    }
}
